cmake/macros/macro_evaluate_expression.cmake

```diff
@@ -0,0 +1,31 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2012 - 2017 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# A macro that evaluates an expression as supplied by a string. Suggestion
+# from the Wiki http://cmake.org/Wiki/CMake/Language_Syntax
+#
+# USAGE:
+#
+# EVALUATE_EXPRESSION("<expression>")
+#
+
+MACRO(EVALUATE_EXPRESSION _the_expression)
+  SET(_tmp_name
+    "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CMakeTmp/evaluate_expression.tmp"
+    )
+  FILE(WRITE ${_tmp_name} "${_the_expression}")
+  INCLUDE("${_tmp_name}")
+ENDMACRO()
```

Comment:
Nice -- a run-time evaluator. Almost LISP-like.

Would it be worth (i) deleting the file again at the end, (ii) using `${CMAKE_CURRENT_BINARY_DIR}` to avoid different macros stepping on each other's files? Or is this what `${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}` does?